tests/EppoClientTest.php

```diff
@@ -267,11 +265,37 @@ public function testHandlesLoggingException()
         $this->assertEquals('control', $assignment);
     }
 
+    public function testGracefulModeDoesNotThrow()
+    {
+        $pollerMock = $this->getPollerMock();
+        $mockConfigRequester = $this->getExperimentConfigurationRequesterMock(self::MOCK_EXPERIMENT_CONFIG, new Exception('config requester error'));
+        $mockLogger = $this->getMockBuilder(LoggerInterface::class)->getMock();
+
+        $subjectAttributes = [['foo' => 3]];
+        $client = EppoClient::createTestClient($mockConfigRequester, $pollerMock, $mockLogger, true);
+
+        $assignment = $client->getAssignment('subject-10', self::EXPERIMENT_NAME, $subjectAttributes);
+        $this->assertNull($assignment);
+    }
+
+    public function testNoGracefulModeThrows()
+    {
+        $pollerMock = $this->getPollerMock();
+        $mockConfigRequester = $this->getExperimentConfigurationRequesterMock(self::MOCK_EXPERIMENT_CONFIG, new Exception('config requester error'));
+        $mockLogger = $this->getMockBuilder(LoggerInterface::class)->getMock();
+
+        $subjectAttributes = [['foo' => 3]];
+        $client = EppoClient::createTestClient($mockConfigRequester, $pollerMock, $mockLogger, false);
+
+        $this->expectException(Exception::class);
+        $client->getAssignment('subject-10', self::EXPERIMENT_NAME, $subjectAttributes);
+    }
+
```

Comment:
new tests - do you want to see these extended to all the public methods?

Reply:
I think it would be worth it to be thorough as long as it could be done in a way that reused 95% of the code